pubsub.go

```diff
@@ -475,6 +485,16 @@ func WithMaxMessageSize(maxMessageSize int) Option {
 	}
 }
 
+// WithProtocolMatchFn sets a custom matching function for protocol selection to
+// be used by the protocol handler on the Host's Mux. Can be combined with
+// WithGossipSubProtocols feature function for checking if a protocol is supported
```

Comment:
I would say here "Should be combined ... for checking if certain protocol features are supported"